Create ProcessForCreatingUpdatingDeletingAPIDescriptionsAtTheWebsite (#16)
camaraproject/Marketing: documentation/SupportingDocuments/ProcessForCreatingUpdatingDeletingAPIDescriptionsAtTheWebsite.md

# Process for creating, updating and deleting API descriptions at the CAMARA website

## Process for content creation:

> 

Reviewer: ```suggestion
Actor: Sub Project or Independent Sandbox team

```

- New API (name)

>   - Trigger: Sub Project / Independent Sandbox decides about (stable!) API name

Reviewer: ```suggestion
  - Trigger: Actor finalizes the decision about  API name
```

    - Action: Sub Project / Independent Sandbox creates API description (based on given Wiki template) and creates issue in Marketing repo (referencing Wiki page)
- New / update API public release
  - Trigger: Sub Project / Independent Sandbox creates a new/update public release of API YAML

>     - Action: Sub Project / Independent Sandbox creates issue in Marketing repo with updates to API description (referencing updated Wiki page) and release (referencing release tracker)

Reviewer: ```suggestion
    - Action: Sub Project / Independent Sandbox creates issue in Marketing repo to inform about the API description update (referencing updated Wiki page) and release (referencing release tracker)
```

- API public release becomes part of a meta-release
  - Trigger: Sub Project / Independent Sandbox applies for API being part of a meta-release (M3/M4)

>     - Action: Sub Project / Independent Sandbox creates issue in Marketing repo with updates to API description (referencing updated Wiki page) and release (referencing release tracker)

Reviewer: ```suggestion
    - Action: Sub Project / Independent Sandbox creates issue in Marketing repo to inform about the API description update (referencing updated Wiki page) and release (referencing release tracker)
```

- Change in API description requested

>   - Trigger: Change request (by anybody)

Reviewer: ```suggestion
  - Trigger: Change request (by anybody, anywhere)
```

    - Action: Forward change request to Sub Project / Independent Sandbox with reminder to check/update API description
    - Action: Sub Project / Independent Sandbox creates issue in Marketing repo with updates to API description (referencing updated Wiki page)
- Delete API
  - Trigger: TSC archives API repository
    - Action: TSC creates issue in Marketing repo

## Process for website update:

- Website update

>   - Trigger: Marketing team gets issue

Reviewer: ```suggestion
  - Trigger: Marketing team gets issue informing of API description update
```

  - Trigger: Marketing team performs cross-check with meta-release API list

>     - Action: Marketing team updates API description / release information at website

Reviewer: ```suggestion
    - Action: Marketing team updates API description / release information at website using updated information received
```

>     - Action: Marketing team asks Sub Project / Independent Sandbox, API Backlog Working Group and OGW Product Workstream to review

Reviewer: ```suggestion
    - Action: Marketing team asks Sub Project / Independent Sandbox, API Backlog Working Group and OGW Product Workstream to review website content
```

>     - Action: Marketing team includes review feedback (at website and Wiki page)

Author: In 99% of the cases only a small change is requested. But for sure if it is something bigger / critical we go back to the actor.

# Process for creating, updating and deleting API descriptions at the CAMARA website

> ## Process for content creation:

Reviewer 2: I wouldn't call the process steps again a process, it's confusing.

```suggestion
## Step1: Content creation within CAMARA wiki and trigger to Marketing Working Group
```

Actor: Sub Project or Independent Sandbox team

- New API (name)
  - Trigger: Actor finalizes the decision about the API name

>     - Action: Actor creates API description (based on given [Wiki template](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/110854523/API+NAME+API+description)) and creates issue in Marketing repo (referencing Wiki page)

Reviewer 3: Before the content is published on the website, is there any technical or editorial validation?
```suggestion
    - Action (pre-website update): Actor (Sub Project / Sandbox team) ensures the description complies with current structure guidelines (template, tone, terminology).
    - Action: Actor creates API description (based on given [Wiki template](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/110854523/API+NAME+API+description)) and creates issue in Marketing repo (referencing Wiki page)
```

- New / update API public release
  - Trigger: Actor creates a new/update public release of API YAML
    - Action: Actor creates issue in Marketing repo to inform about the API description update (referencing updated Wiki page) and release (referencing release tracker)
- API public release becomes part of a meta-release
  - Trigger: Actor applies for API being part of a meta-release (M3/M4)
    - Action: Actor creates issue in Marketing repo to inform about the API description update (referencing updated Wiki page) and release (referencing release tracker)
- Change in API description requested
  - Trigger: Change request (by anybody, anywhere)
    - Action: Forward change request to Actor with reminder to check/update API description
    - Action: Actor creates issue in Marketing repo with updates to API description (referencing updated Wiki page)

> - Delete API
>   - Trigger: TSC archives API repository

Reviewer 2: I would generalize this to API Repository stage changes, which are decided within TSC (and API will never be deleted, they will get "Archived"). 

```suggestion
- Change of maturity stage of an API Repository: Sandbox -> Incubating -> Graduated (or -> Archived)
  - Trigger: TSC decides about an API Repository transition
```

    - Action: TSC creates issue in Marketing repo

> 

Reviewer 3: Is there a process for revising old descriptions that are already published? We could add something like:
```
Trigger: Scheduled content review (each meta-release)
Action: Actor checks that API descriptions reflect the current state of the API and references are up to date.
```

Reviewer 2: As discussed we anchor within the release review process at M4 the confirmation from the actor, that they have reviewed the (existing) API description. They have to confirm that in the API Readiness Checklist.

Within in the preparation of a meta-release (M5 milestone) the web pages of all APIs which are participating in the meta-release need to be updated (at least the API version). That should be done in a batch process, not via single issues. @MarkusKuemmerle would you like to add this process to the description?

> ## Process for website update:

Reviewer 2: ```suggestion
## Step 2: Update of web site by Marketing Working Group
```

Actor: Marketing team

> 

Reviewer 3: It would be nice if after the web update, the Marketing team would share the link to the updated page? We should put it as an option

Reviewer 2: @albertoramosmonagas the link was already shared in the review step, which additional sharing do you see here?

- Website update
  - Trigger: Actor gets issue informing of API description update

>   - Trigger: Actor performs cross-check with meta-release API list

Reviewer 2: Not all releases and all information are on a meta-release page (e.g. maturity stage like "Incubating").

```suggestion
  - Trigger: Actor performs cross-check with meta-release API list and meta-data on API repository wiki page
```
